core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1681,6 +1675,17 @@ private void unparseRegexMatchCount(SqlWriter writer, SqlCall call,
     writer.endFunCall(arrayLengthFrame);
   }
 
+  private void unParseTimestampSeconds(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec,
+      SqlFunction sqlFunction) {
+    if (call.operandCount() == 1) {
+      castAsDatetime(writer, call, leftPrec, rightPrec, sqlFunction);
+    } else {
+      SqlCall timestampSecondsCall = sqlFunction.createCall(SqlParserPos.ZERO,
```

Comment:
This behavior should only be there in case second operand is true.

Reply:
resolved